charts/opentelemetry-demo/values.yaml

```diff
@@ -500,14 +516,112 @@ opentelemetry-collector:
               allowed_origins:
                 - "http://*"
                 - "https://*"
+      otlp/spanmetrics:
+        protocols:
+          grpc:
+            endpoint: "localhost:65535"
+      prometheus:
+        config:
+          scrape_configs:
+            - job_name: 'otel-collector'
+              scrape_interval: 5s
+              static_configs:
+                - targets: ['0.0.0.0:8888']
+
     exporters:
       otlp:
-        endpoint: '{{ .Release.Name }}-jaeger:4317'
+        endpoint: '{{ include "otel-demo.name" . }}-jaeger:4317'
         tls:
           insecure: true
+      prometheus:
+        endpoint: '0.0.0.0:9464'
+
+    processors:
+      spanmetrics:
+        metrics_exporter: prometheus
+
     service:
       pipelines:
         traces:
-          exporters:
-            - logging
-            - otlp
+          receivers: [otlp]
+          processors: [memory_limiter, spanmetrics, batch]
+          exporters: [otlp, logging]
+        metrics:
+          receivers: [otlp, prometheus]
+          exporters: [prometheus, logging]
+        metrics/spanmetrics:
+          receivers: [otlp/spanmetrics]
+          processors: [memory_limiter, batch]
+          exporters: [prometheus, logging]
```

Comment:
Why do we need this pipeline? It doesn't seem to be used...

Reply:
This is how we have the collector configured to generate RED metrics from all spans, and send them into Prometheus. This is a copy from what we have for the collector running in the demo repo's config https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/otelcollector/otelcol-config.yml

Reply:
> This is how we have the collector configured to generate RED metrics from all spans, and send them into Prometheus.

I see that spanmetrics receiver is configured to send metrics to prometheus exporter, can please you point me to a source that is sending otlp metric to localhost:65535?

Reply:
This is part of how you configure the spanmetrics receiver/processor to work. It needs a dummy receiver to set up a pipeline, so we set it to an endpoint that should not be used for anything else.

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/spanmetricsprocessor

Reply:
Submitted a PR for the demo repo https://github.com/open-telemetry/opentelemetry-demo/pull/553

Reply:
The spanmetrics processor is absolutely being used in the demo.  This is a screenshot of some of the metrics it generates in Grafana.
![Screen Shot 2022-11-02 at 8 27 20 PM](https://user-images.githubusercontent.com/1296118/199626647-d380b0f9-14b0-49e2-9353-b80cbb57444c.png)

Reply:
I'm not saying that spanmetrics processor is not needed

Reply:
The way the spanmetrics processor works is it requires a receiver and its own pipeline to generate metrics from traces and send them along.  We have it configured properly per the docs, and it does work in our demo as shown here.

Reply:
Can you remove this pipeline and try again? I'm pretty sure it's not needed

Reply:
I understand the mis-communication here now.  The spanmetrics processor was using the prometheus receiver, and not the otlp/spanmetrics one. That must of changed on the demo side some time ago and I missed it.

I did try without the otlp/spanmetrics receiver and pipeline, and this is working as expected.

Reply:
Ya the spanmetricsprocessor readme is a little misleading (which is what I think prompted this extra pipeline in the demo)